FirebaseAuth/Sources/MultiFactor/FIRMultiFactorSession.m

```diff
@@ -42,6 +42,7 @@ + (FIRMultiFactorSession *)sessionForCurrentUser {
   FIRUser *currentUser = [[FIRAuth auth] currentUser];
   NSString *IDToken = currentUser.rawAccessToken;
   FIRMultiFactorSession *session = [[FIRMultiFactorSession alloc] initWithIDToken:IDToken];
+  session.currentUser = currentUser;
```

Comment:
The PhoneAuthProvider in case of phone has an auth variable instance to set the request configuration. [link](https://github.com/firebase/firebase-ios-sdk/blob/totp-enrollment/FirebaseAuth/Sources/AuthProvider/Phone/FIRPhoneAuthProvider.m#L488)
For TOTP, we need to set it within the generateSecret function, so the user's auth instance needs to be passed there. [link](https://github.com/firebase/firebase-ios-sdk/blob/totp-enrollment/FirebaseAuth/Sources/MultiFactor/TOTP/FIRTOTPMultiFactorGenerator.m#L42)